taf/developer_tool.py

```diff
@@ -40,21 +40,25 @@ def add_target_repos(repo_path, targets_directory, namespace=''):
     namespace:
       Namespace used to form the full name of the target repositories. E.g. some_namespace/law-xml
   """
-  auth_repo_targets_dir = os.path.join(repo_path, TARGETS_DIRECTORY_NAME)
+  repo_path = Path(repo_path).resolve()
+  targets_directory = Path(targets_directory).resolve()
+  if namespace is None:
+    namespace = targets_directory.name
+  auth_repo_targets_dir = repo_path / TARGETS_DIRECTORY_NAME
   if namespace:
-    auth_repo_targets_dir = os.path.join(auth_repo_targets_dir, namespace)
-    if not os.path.exists(auth_repo_targets_dir):
+    auth_repo_targets_dir = auth_repo_targets_dir / namespace
+    if not auth_repo_targets_dir.exists():
       os.makedirs(auth_repo_targets_dir)
 
-  for target_repo_dir in os.listdir(targets_directory):
-    repo_path = os.path.join(targets_directory, target_repo_dir)
-    if os.path.isdir(repo_path):
-      target_repo = GitRepository(os.path.join(targets_directory, target_repo_dir))
-      if target_repo.is_git_repository:
-        commit = target_repo.head_commit_sha()
-        target_repo_name = os.path.basename(target_repo_dir)
-        with open(os.path.join(auth_repo_targets_dir, target_repo_name), 'w') as f:
-          json.dump({'commit': commit}, f,  indent=4)
+  for target_repo_dir in targets_directory.glob('*'):
+    if not target_repo_dir.is_dir() or target_repo_dir == repo_path:
+      continue
+    target_repo = GitRepository(str(target_repo_dir))
+    if target_repo.is_git_repository:
+      commit = target_repo.head_commit_sha()
+      target_repo_name = os.path.basename(target_repo_dir)
+      with open(os.path.join(auth_repo_targets_dir, target_repo_name), 'w') as f:
```

Comment:
`(auth_repo_targets_dir / target_repo_name).write_text(json.dumps({'commit': commit}, indent=4))`, but if you want to keep `with open`, you can just replace `os.path.join` here.

Reply:
Using `write_text` here seems fine here, will also update `generate_repositories_json`